frontend/src/lib/notification.tsx

```diff
@@ -8,15 +9,22 @@ export class Notification {
   url?: string;
   date: number | string = new Date().getTime();
   deleted: boolean = false;
+  cluster: string | null = getCluster();
 
-  constructor(message?: string, date?: number | string) {
-    if (message) {
-      this.message = message;
+  constructor(options?: { message?: string; date?: number | string; cluster?: string | null }) {
+    if (options) {
+      const { message, date, cluster } = options;
+      if (message) {
+        this.message = message;
+      }
+      if (date) {
+        this.date = date;
+      }
+      if (cluster) {
+        this.cluster = cluster;
+      }
+      // generate the id based on the message and the date attached to a notification
     }
-    if (date) {
-      this.date = date;
-    }
-    // generate the id based on the message and the date attached to a notification
     this.id = btoa(unescape(encodeURIComponent(`${this.date},${this.message}`)));
```

Comment:
Would cluster be needed in making the ID too?

Reply:
Good catch. I think it's a good idea.

Reply:
updated